pkg/reconciler/buildrun/resources/sources.go

```diff
@@ -17,8 +17,14 @@ func AmendTaskSpecWithSources(
 	taskSpec *v1beta1.TaskSpec,
 	build *buildv1alpha1.Build,
 ) {
-	// create the step for spec.source, this is always Git
-	sources.AppendGitStep(cfg, taskSpec, build.Spec.Source, "default")
+	// create the step for spec.source, either Git or Bundle
+	switch {
+	case build.Spec.Source.Container != nil:
+		sources.AppendBundleStep(cfg, taskSpec, build.Spec.Source, "default")
+
+	default:
```

Comment:
This will allow a Build to have no sources, which is imo valid (considering a build strategy that does `crane rebase` and expects its input, source image, old and new base image, through `params`).

```suggestion
	case build.Spec.Source.URL != "":
```